packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -55,6 +55,28 @@ These are internal properties on the column. They will be set by a concrete impl
 | operandDataRecordFieldName | `string` or `undefined`                                                                                                     | The name of the data field that will be used for operations on the table, such as sorting and grouping |
 | sortOperation              | `TableColumnSortOperation`, initially defined as `{ basic: 'basic', localeAwareCaseSensitive: 'localeAwareCaseSensitive' }` | The operation to use for sorting                                                                       |
 
+#### Events
+
+After any interactive sort operation, a `column-sort-change` event will be fired, with event details of type `TableColumnSortEventDetail`:
+
+```ts
+interface TableColumnSortState {
+    columnId: string;
+    sortIndex: number;
+    sortDirection: TableColumnSortDirection;
+}
+
+export interface TableColumnSortEventDetail {
+    sortedColumns: TableColumnSortState[];
```

Comment:
Is there any value in exposing `sortedColumns` as a property on the table too so that someone can read its value at times other than when the event fires?

This is more about API completeness rather than a concrete use case, so we can always add it later. But I could imagine a use case where a client triggers a new async data fetch in response to the sort change event but then needs to know the sort state again when the fetch completes. (That's where my use case breaks down; I'm not sure *why* they'd want the sort state. Maybe to validate the response or something?)

Reply:
Yeah I can't really think of a specific use case for it either, so I'd default to not adding it for now.